src/err/mod.rs

```diff
@@ -65,15 +65,15 @@ impl<'a> PyDowncastError<'a> {
 
 /// Error that indicates a failure to convert a PyAny to a more specific Python type.
 #[derive(Debug)]
-pub(crate) struct PyDowncastError2<'a, 'py> {
-    from: &'a Py2<'py, PyAny>,
+pub struct PyDowncastError2<'a, 'py> {
```

Comment:
Oh, this one needs a matching name as well. `PyBoundDowncastError`?

Reply:
I've added an item for #3684, to remember this (and also the iterator types), if we want to save this for a follow-up? Maybe just `DowncastError` and `DowncastIntoError`? Could also rename the gil-ref one to `PyAnyDowncastError` and make this `PyDowncastError`?

Reply:
I think it would like concentrate the renaming here because I mainly just want to be done with it.

> Maybe just DowncastError and DowncastIntoError?

Are my favourites from the list. I would not rename the existing type as it appears as gratuitous churn to me.